# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../kdemultimedia/wordsize_check.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -Naur kdemultimedia-3.4.0.orig/mpeglib/lib/input/cdromAccess.cpp kdemultimedia-3.4.0/mpeglib/lib/input/cdromAccess.cpp
--- kdemultimedia-3.4.0.orig/mpeglib/lib/input/cdromAccess.cpp	2005-01-03 16:53:36.000000000 +0200
+++ kdemultimedia-3.4.0/mpeglib/lib/input/cdromAccess.cpp	2005-03-06 15:04:12.252743928 +0200
@@ -29,7 +29,11 @@
   #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,70)
   #include <bits/wordsize.h>
   #if __WORDSIZE == 64
-  typedef unsigned long __u64;
+  #ifdef __x86_64__
+	  typedef unsigned long long __u64;
+  #else
+	  typedef unsigned long __u64;
+  #endif
   #else
   typedef unsigned long long __u64;
   #endif 
